management/server/idp/auth0.go

```diff
@@ -183,6 +186,136 @@ func (c *Auth0Credentials) Authenticate() (JWTToken, error) {
 	return c.jwtToken, nil
 }
 
+func batchRequestUsersUrl(authIssuer, accountId string, page int) (string, url.Values, error) {
+	u, err := url.Parse(authIssuer + "/api/v2/users")
+	if err != nil {
+		return "", nil, err
+	}
+	q := u.Query()
+	q.Set("page", strconv.Itoa(page))
+	q.Set("search_engine", "v3")
+	q.Set("q", "app_metadata.wt_account_id:"+accountId)
+	u.RawQuery = q.Encode()
+
+	return u.String(), q, nil
+}
+
+func requestByUserIdUrl(authIssuer, userId string) string {
+	return authIssuer + "/api/v2/users/" + userId
+}
+
+// Requests users in batches from Auth0
+func (am *Auth0Manager) GetBatchedUserData(accountId string) ([]*UserData, error) {
+	jwtToken, err := am.credentials.Authenticate()
+	if err != nil {
+		return nil, err
+	}
+
+	var list []*UserData
+
+	// https://auth0.com/docs/manage-users/user-search/retrieve-users-with-get-users-endpoint#limitations
+	// auth0 limitation of 1000 users via this endpoint
+	for page := 0; page < 20; page++ {
+		url, query, err := batchRequestUsersUrl(am.authIssuer, accountId, page)
+		if err != nil {
+			return nil, err
+		}
+
+		req, err := http.NewRequest(http.MethodGet, url, strings.NewReader(query.Encode()))
+		if err != nil {
+			return nil, err
+		}
+
+		req.Header.Add("authorization", "Bearer "+jwtToken.AccessToken)
+		req.Header.Add("content-type", "application/json")
+
+		res, err := am.httpClient.Do(req)
+		if err != nil {
+			return nil, err
+		}
+
+		body, err := io.ReadAll(res.Body)
+		if err != nil {
+			return nil, err
+		}
+
+		var batch []UserData
+		err = json.Unmarshal(body, &batch)
+		if err != nil {
+			return nil, err
+		}
+
+		log.Debugf("Requested batch; %v", batch)
+
+		defer func() {
+			err = res.Body.Close()
+			if err != nil {
+				log.Errorf("error while closing update user app metadata response body: %v", err)
+			}
+		}()
+
+		if len(batch) == 0 {
```

Comment:
this check should be performed after StatusCode as a non 200 status will cause len(batch) == 0

Reply:
Ah thanks, good point